### PR TITLE
Added translation support.

### LINKED ISSUE #20: Fix translation support
Today, core uses the `t()` function of the framework interface with a default english translation.

We will have to change how the translation works in Core. We should probably use keys and then allow the implementation to create/use a map inside the `t()` function to fetch the correct translation. This will probably be the easiest way to deal with translations for most web systems.

### PR DESCRIPTION
**Pull request by thomasmars**
Added a mapping to the translation files.
Fixes #20

### CODE REVIEW

**Comment by juho-jaakkola** [comment]
I dislike that fact that this mixes up two different string substitution syntaxes in the language file.

**Comment by juho-jaakkola** [comment]
How about using Moodle's [own syntax](https://docs.moodle.org/dev/String_API#get_string.28.29) in `lang/<key>/hvp.php`, and adding logic for converting the placeholders:

Before/after:
```
'Illegal option %option in %library';
'Illegal option {$a->option} in {$a->library}';

**Comment by thomasmars** [comment]
Sure, that makes sense. I'll have a look at it

File: lang/en/hvp.php [hunk 1]
```diff
@@ -132,3 +132,57 @@
...
+$string['updatedlibraries'] = 'Updated %old H5P libraries.';
+$string['missingdependency'] = 'Missing dependency @dep required by @lib.';
+$string['invalidstring'] = 'Provided string is not valid according to regexp in semantics. (value: \"%value\", regexp: \"%regexp\")';
+$string['invalidfile'] = 'File "%filename" not allowed. Only files with the following extensions are allowed: %files-allowed.';
+$string['invalidmultiselectoption'] = 'Invalid selected option in multi-select.';
+$string['invalidselectoption'] = 'Invalid selected option in select.';
+$string['invalidsemanticstype'] = 'H5P internal error: unknown content type "@type" in semantics. Removing content!';
+$string['invalidsemantics'] = 'Library used in content is not a valid library according to semantics';
+$string['copyrightinfo'] = 'Copyright information';
+$string['titole'] = 'Title';
```
Comment (icc) [inline lang/en/hvp.php]: titole? :-)

File: classes/framework.php [hunk 2]
```diff
@@ -176,10 +176,77 @@ public static function printMessages($type, $messages) {
...
+                'Unable to create directory.' => get_string('unabletocreatedir', 'hvp'),
+                'Unable to get field type.' => get_string('unabletogetfieldtype', 'hvp'),
+                "File type isn't allowed." => get_string('filetypenotallowed', 'hvp'),
+                'Invalid field type.' => get_string('invalidfieldtype', 'hvp'),
+                'Invalid image file format. Use jpg, png or gif.' => get_string('invalidimageformat', 'hvp'),
+                'File is not an image.' => get_string('filenotimage', 'hvp'),
+                'Invalid audio file format. Use mp3 or wav.' => get_string('invalidaudioformat', 'hvp'),
+                'Invalid video file format. Use mp4 or webm.' => get_string('invalidvideoformat', 'hvp'),
+                'Could not save file.' => get_string('couldnotsave', 'hvp'),
+                'Could not copy file.' => get_string('couldnotcopy', 'hvp')
```
Comment (icc) [inline classes/framework.php]: Shouldn't the last ten strings be in `lang/en/hvp.php` as well? I get lots of notices.

**Comment by icc** [comment]
Excellent!

**Pull request merged**